ha/core/system_health/system_health.py

```diff
@@ -362,6 +361,26 @@ def _check_and_update(self, current_health: str, updated_health: str, healtheven
         if update_action != HEALTH_EVENT_ACTIONS.IGNORE.value:
             self._update(healthevent, updated_health, next_component=next_component)
 
+    def _get_cvg_list(self, healthevent, match_with):
+        """
+        Parse system health key and get cvg id for matching criteria.
+        Args:
+            healthevent: health event of resource
+            match_with : Additional resource information to be matched with expecting key
+        Returns:
+            List of cvg ids.
+        """
+        cvg_list = []
+        cluster_prefix = self._prepare_key(CLUSTER_ELEMENTS.CLUSTER.value,
+                                           cluster_id=healthevent.cluster_id)
+        cluster_key = _DELIM.join(cluster_prefix.split(_DELIM)[:-1])
+        key_prefix = f"{CLUSTER_CONFSTORE_PREFIX.strip(_DELIM)}{cluster_key}"
+        cvg_list = self.healthmanager.parse_key(CLUSTER_ELEMENTS.CVG.value,
+                                                match_with, key_prefix)
+        if not cvg_list:
+            Log.error(f"Failed to get cvg_id for {healthevent.resource_id} of node {self.node_id}")
+        return cvg_list
```

Comment:
we should get only 1 cvg_id for one node-disk id combination. 
Exception should be raised if we get multiple

Reply:
Yes. But this is a generic function and not only specific to node, disk_id combination. For example, system_health requests all cvg_ids for node, then this will return all cvg_ids.

The return value will be evaluated where this method is called. @ArchanaLimaye

Reply:
Got it. But in that case please add the check at call point.
[We are informed that the disk ids will be unique within a node and we want to catch it early if that assumption gets broken anytime; since all our logic is based on this assumption]

Reply:
Done.